optuna/samplers/_cmaes.py

```diff
@@ -352,38 +358,36 @@ def _restore_optimizer(
 
     def _init_optimizer(
         self,
-        search_space: Dict[str, BaseDistribution],
-        ordered_keys: List[str],
+        trans: _SearchSpaceTransform,
         population_size: Optional[int] = None,
         randomize_start_point: bool = False,
     ) -> CMA:
+
+        lower_bounds = trans.bounds[:, 0]
+        upper_bounds = trans.bounds[:, 1]
+        n_dimension = len(trans.bounds)
```

Comment:
I'm sorry. I found a bug. The shape of `trans.bounds` is `(2, n_dimensions)`. So we need to calculate n_dimension like the following.

```suggestion
        n_dimension = len(lower_bounds)
```